synapse/util/caches/descriptors.py

```diff
@@ -431,6 +431,12 @@ def __get__(
         cache: DeferredCache[CacheKey, Any] = cached_method.cache
         num_args = cached_method.num_args
 
+        if num_args != self.num_args:
+            raise Exception(
+                "Number of args (%s) does not match underlying cache_method_name=%s (%s)."
+                % (self.num_args, self.cached_method_name, num_args)
+            )
```

Comment:
Added a safety check so others don't run into the same pitfall and we can see this error obviously.